simplecpp.cpp

```diff
@@ -1303,21 +1347,26 @@ void simplecpp::TokenList::constFoldLogicalOp(Token *tok)
         }
         if (tok->str() != "&&" && tok->str() != "||")
             continue;
-        if (!tok->previous || !tok->previous->number)
+        if (!tok->previous || !tok->next)
             continue;
-        if (!tok->next || !tok->next->number)
+        if (!tok->previous->number && !tok->next->number)
```

Comment:
I feel it's insufficient to just look at the previous/next token. No folding should occur here:

      X >= 0 || 0 < Y